utils/readiness.go

```diff
@@ -0,0 +1,48 @@
+// Defines an HTTP server that reports the readiness of the ocs-osd-operator,
+// and helper functions to Set/Unset its readiness state.
+// For this to work, ensure the following is added to the Deployment definition
+// in the CSV:
+//                readinessProbe:
+//                  httpGet:
+//                    path: /readyz
+//                    port: 8081
+//                  initialDelaySeconds: 5
+//                  periodSeconds: 10
+
+package utils
+
+import (
+	"errors"
+	"net/http"
+	"sync"
+
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+)
+
+var readinessServer = struct {
+	sync.Mutex
+	ready bool
+}{}
+
+var ReadinessChecker healthz.Checker = func(_ *http.Request) error {
+	readinessServer.Lock()
+	operatorReady := readinessServer.ready
+	readinessServer.Unlock()
+
+	if operatorReady {
+		return nil
+	}
+	return errors.New("Operator not ready yet")
+}
+
+func SetReady() {
+	readinessServer.Lock()
+	readinessServer.ready = true
+	readinessServer.Unlock()
+}
+
+func SetNotReady() {
+	readinessServer.Lock()
+	readinessServer.ready = false
+	readinessServer.Unlock()
+}
```

Comment:
@renan-campos as we agreed when we did the code design discussion, there should be no dependency on shared structures between the controller and the prob. 

All of the above code is unnecessary

Reply:
Addressed in new commit.